marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/matcher-impl/options-impl.xqy

```diff
@@ -452,7 +452,10 @@ declare function opt-impl:compile-match-options(
               if ($is-complex-rule) then
                 $match-rule/(entityPropertyPath|documentXPath)
               else
-                $match-rule/(@property-name|propertyName|((*:all-match|allMatch)/*:property))
+                (
```

Comment:
Looking at the surrounding code, I think it's worthwhile to extract a small function here and add a few unit tests, as this is part of a very large function that has dozens of variable assignments, making it difficult to understand.

This assigns a variable - full-property-node (which possibly should have a plural name since it looks like the assignment can return a sequence?) - that is only used on the next line to get full-property-name. So I think there's a function of:

    get-full-property-name-for-match-rule(match-rule, is-complex)

Some small unit tests could be written against that to verify that it works correctly and, more importantly, to demonstrate some of the possible inputs. For example, the code begs the question - When does all-match/allMatch exist, and does it also exist when property-name/propertyName exists? The test case would answer that question by showing the expected inputs.

Reply:
Added the separate function with unit tests